docs/proposals/backup-restore.md

```diff
@@ -1,131 +1,265 @@
+
+- `--input <path>`: Required. Restore from a folder or archive file to registry.
+- `--exclude-referrers`: Optional. Restore the image from backup excluding referrers.
+- `--dry-run`: Optional. Simulate the restore process without actually uploading any artifacts to the target registry.
+
+**Common flags:**
+
+* `--concurrency <int>`: Number of parallel upload operations. Default: `3`.
+* `--plain-http`: Allow insecure connections to registry without SSL check.
+* `--insecure`: Allow connections to registries without valid TLS certificates.
+* `--registry-config <path>`: Path to the authentication configuration file for the registry.
+* `--username <string>`: Username for authenticating to the registry.
+* `--password <string>`: Password for authenticating to the registry.
+* `--password-stdin`: Read password from stdin.
+* `--identity-token <string>`: Use bearer token for authentication.
+* `--identity-token-stdin`: Read identity token from stdin.
+* `--ca-file <path>`: Path to custom CA certificate file.
+* `--cert-file <path>`: Path to client TLS certificate file.
+* `--key-file <path>`: Path to client private key file.
+* `--resolve <host:port:address[:address_port]>`: Customized DNS for registry.
+* `--debug`: Output debug logs (implies `--no-tty`).
+* `--distribution-spec string`: [Preview] set OCI distribution spec version and API option for target. Options: v1.1-referrers-tag, v1.1-referrers-api
+* `--no-tty`: Disable progress bars.
+
+### User Experience in the CLI
+
+The desired end-to-end user experience of using `oras backup` and `oras restore` to address the identified problems and support the outlined user scenarios is illustrated below.
 
-The output directory structure is a single OCI layout containing all of the artifacts.
-Each artifact in the output OCI layout will be tagged with the name of source.
-For example, the `registry.k8s.io/kube-apiserver-arm64:v1.31.0` artifact will be tagged `registry.k8s.io/kube-apiserver-arm64:v1.31.0`.
+#### Backup an Entire Repository to a Tarball and Restore to Another Registry
+
+Assume two tags `v1` and `v2` are stored in a repository `registry.k8s.io/kube-apiserver` and each tag has one referrer. Backup the entire repo to a tarball and restore it to another registry:
 
 ```bash
-% oras repo tags --oci-layout ./mirror
-registry.k8s.io/kube-apiserver-arm64:v1.31.0
-registry.k8s.io/kube-apiserver-arm64:v1.32.0
-registry.k8s.io/kube-controller-manager-arm64:v1.31.0
+## Backup a repository from a registry to a local tarball. All tags and their referrers will be included.
+oras backup --output backup.tar --include-referrers registry-a.k8s.io/kube-apiserver
 ```
 
-The backup command will also have the ability to write output to a new compressed tar file where the contents are in a single oci-layout
-format. For example:
+Alternatively, `oras` should also enables users to choose which tags within a repository to back up. For example, back up `v1` and `v2` tags with their referrers:
```

Comment:
Do we support specifying one or more digests as well? Is it possible to do something like: 
```bash
oras backup registry-a.k8s.io/kube-apiserver:v1,@abc123,@def456
``` 
?

If so, we need to put some examples here.

Reply:
I don't think we should support digests here. The `oras backup` and `oras restore` are to cover the most common scenarios and we should leave edge scenarios to `oras copy`.

Reply:
@Wwwsylvia  It makes sense to support with backup and restore tags in the 1st iteration as tags are used the most for image manifest and image index. Digest-based backup and restore could be considered in v1.4.0-alpha if we received users inquires.